src/common/src/util/resource_util.rs

```diff
@@ -0,0 +1,624 @@
+                };
+                assert_eq!(
+                    read_integer_from_file_path(test_file_path),
+                    curr_test_case.expected
+                );
+                if Path::new(test_file_path).exists() {
+                    fs::remove_file(test_file_path).expect("File delete failed");
+                }
+            }
+        }
+
+        #[test]
+        fn test_parse_cgroup_v2_cpu_limit_string() {
+            struct TestCase {
+                file_exists: bool,
+                value_in_file: String,
+                expected: Option<f32>,
+            }
+
+            let test_cases = HashMap::from([
+                (
+                    "valid-values-in-file",
+                    TestCase {
+                        file_exists: true,
+                        value_in_file: String::from("10000 20000"),
+                        expected: Some(10000.0 / 20000.0),
+                    },
+                ),
+                (
+                    "Invalid-single-value-in-file",
+                    TestCase {
+                        file_exists: true,
+                        value_in_file: String::from("10000"),
+                        expected: None,
+                    },
+                ),
+                (
+                    "Invalid-string-value-in-file",
+                    TestCase {
+                        file_exists: true,
+                        value_in_file: String::from("10000 test-string "),
+                        expected: None,
+                    },
+                ),
+                (
+                    "max-value-in-file",
+                    TestCase {
+                        file_exists: true,
+                        value_in_file: String::from("max 20000"),
+                        expected: None,
+                    },
+                ),
+                (
+                    "file-not-exist",
+                    TestCase {
+                        file_exists: false,
+                        value_in_file: String::from(""),
+                        expected: None,
+                    },
+                ),
+            ]);
+
+            let test_file_path = "resource-util-test-2";
+            for tc in test_cases {
+                let curr_test_case = &tc.1;
+                if curr_test_case.file_exists {
+                    let mut file = File::create(test_file_path)
+                        .expect("Error encountered while creating file!");
+                    file.write_all(curr_test_case.value_in_file.as_bytes())
+                        .expect("Error while writing to file");
+                };
+                assert_eq!(
+                    read_cgroup_v2_cpu_limit_from_file_path(test_file_path),
+                    curr_test_case.expected
+                );
+                if Path::new(test_file_path).exists() {
+                    fs::remove_file(test_file_path).expect("File delete failed");
+                }
```

Comment:
You can use the [`tempfile`](https://docs.rs/tempfile/3.3.0/tempfile/struct.NamedTempFile.html#method.as_file_mut) library to deal with it. It could delete the file automatically.

```toml
# Add this to Cargo.toml
[dev-dependencies]
tempfile = "3"
```

```suggestion
                if curr_test_case.file_exists {
                    let mut file = tempfile::NamedTempFile::new()
                        .expect("Error encountered while creating file!");
                    file.as_file_mut()
                        .write_all(curr_test_case.value_in_file.as_bytes())
                        .expect("Error while writing to file");
                    assert_eq!(
                        read_cgroup_v2_cpu_limit_from_file_path(file.path()),
                        curr_test_case.expected
                    );
                };
```